### ISSUE TITLE
GitLab returns 500

### ISSUE THREAD

**Issue by hermanzdosilovic** [timoschwarzer/gitlab-monitor#59]
Hi @timoschwarzer,

Do you have any idea why GitLab returns `500`  on this API request: 
`GET https://gitlab.com/api/v4/projects?order_by=last_activity_at&per_page=20&visibility=private`?

This is configuration I am using for your monitor:
```
{
  "gitlabApi": "https://gitlab.com/api/v4",
  "privateToken": "<MY PRIVATE API TOKEN>",
  "maxAge": 0,
  "fetchCount": 20,
  "pipelinesOnly": false,
  "autoZoom": false,
  "showPipelineIds": true,
  "showJobs": "icon",
  "showDurations": true,
  "showUsers": false,
  "projectVisibility": "private",
  "title": null,
  "filter": {
    "include": ".*",
    "includeTags": ".*",
    "exclude": null,
    "excludeTags": null
  },
  "projectFilter": {
    "*": {
      "include": ".*",
      "exclude": null,
      "default": null,
      "showMerged": true,
      "showTags": true
    }
  }
}
```
Your monitor loads quite long and when I inspected network requests I saw that for this configuration your monitor is calling above given API point. I tried calling the same API point from Postman and I am getting again `500`. My API token is valid, I checked that by listing my private groups: `GET https://gitlab.com/api/v4/groups?owned=yes`.

So basically, I cannot list private projects from my private group with a valid API private token and thus I cannot see anything on your monitor.

Note that I also have one or more private projects that belong to my account on GitLab. I also have one or more private projects in my private group.

Thanks for any help and thanks for your awesome open-source contribution!

Best regards,
Herman

**Comment by timoschwarzer** [comment]
Hi Herman!

Does the response that comes with the 500-error contain any additional information?
This is most likely a problem on GitLab's side. (500 == Internal Server Error)

**Comment by timoschwarzer** [comment]
I just tried your exact config with my own GitLab instance and it works just fine.
The only difference is that they currently run GitLab 11.8.0-rc5 on gitlab.com and I run 11.7.5 on my personal instance. (See https://gitlab.com/api/v4/version)
So my guess is that the API is broken on 11.8.0-rc5.
I'll leave this issue open so we can check back later and see if the issue got fixed :wink:

**Comment by hermanzdosilovic** [comment]
No additional information. Just status 500 with this in body:
```
{
    "message": "500 Internal Server Error"
}
```

But I tried again now and I got expected response back (after ~30s) and then I tried again and got 500 again.

**Comment by hermanzdosilovic** [comment]
I know that is their "fault", but I was just wondering if you have any experience with their API (not)working. :smile:

Thanks for your fast response. I guess I need to wait until they fix this problem on their side. :smiley:

**Comment by timoschwarzer** [comment]
> But I tried again now and I got expected response back (after ~30s) and then I tried again and got 500 again.

Yep, that's definitely not intended. :smile: 

> I was just wondering if you have any experience with their API (not)working

No, since I use my own instance and didn't have any problems with it yet. :wink:

**Comment by timoschwarzer** [comment]
Does this still happen? (They updated to 11.8)

**Comment by hermanzdosilovic** [comment]
Noup, it doesn't still work. But in my private fork I have added [here](https://github.com/timoschwarzer/gitlab-monitor/blob/3ba8cca95ede2fc69a595d02e4f4f2aa19bc495e/src/components/app.vue#L79-L83) new flag `owned=true` so now this part looks like this:

```javascript
const gitlabApiParams = {
    order_by: 'last_activity_at',
    // GitLab per_page max is 100. We use > 100 values as next page follow trigger
    per_page: fetchCount > 100 ? 100 : fetchCount,
    owned: true
}
```

And it works as expected! :)

**Comment by timoschwarzer** [comment]
@hermanzdosilovic Hmm, thank you for the information.

Though adding `owned: true` works, it might filter out some projects which is bad. I think the only thing we can do is waiting for GitLab to fix it.

**Comment by timoschwarzer** [comment]
Closing for now. Feel free to reopen.

**Comment by beanaroo** [comment]
Hi @timoschwarzer, do we have a link to the relevant GitLab issue? From what I can tell, the request is cancelled after 15 seconds. Is the default query perhaps trying to resolve all projects hosted in gitlab.com?

**Comment by timoschwarzer** [comment]
@beanaroo I couldn't reproduce the issue at all. So I don't really know what could be the issue here...
